### PR TITLE
`Development`: Fix docker command in Jenkins development setup

### PR DESCRIPTION
**Pull request by reschandreas**
#### General

- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context

I followed all steps of the setup guide in the documentation, but docker did not work for me during the execution of jobs for programming exercises (I'm on mac). This PR fixes that by installing docker in the container instead of mounting the executable into it, as the socket is still mounted it is now possible to execute docker in the Artemis jenkins image.

It would be great if I could get a review from people using:
- [ ] MacOS
- [ ] Windows
- [ ] Linux 🐧
### Steps for Testing

Prerequisites:
- GitLab and Jenkins Dev Setup

1. Log in to Artemis
2. Create a programming exercise and check if the solution build is green, meaning that the docker image in the build plan was executed correctly.

No need to deploy it, it only affects/fixes your local setup.

#### Performance Review
- [ ] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [ ] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### CODE REVIEW

**Comment by b-fein** [comment]
Not tested, but one thing that should be tested: The Docker binary installed inside the container might not be able to talk to the passed-through Docker socket in case there is a version mismatch, e.g., in case on a Linux system an older version of Docker is installed from the distribution repositories.

File: docker/jenkins/Dockerfile [hunk 1]
```diff
@@ -12,4 +12,16 @@ RUN jenkins-plugin-cli --plugin-file /usr/share/jenkins/ref/plugins.yml
 # (Uncomment this line if you're using jenknis-casc-config.yml for auto-configuration)
 # ENV JAVA_OPTS -Djenkins.install.runSetupWizard=false
 
+# setup docker
+RUN apt-get update  \
```
Comment (Ceiridge) [inline docker/jenkins/Dockerfile:16]: It is recommended to use `apt-get -y update` in scripts to disable interactive prompts when updating

File: docker/jenkins/Dockerfile [hunk 2]
```diff
@@ -12,4 +12,16 @@ RUN jenkins-plugin-cli --plugin-file /usr/share/jenkins/ref/plugins.yml
...
+    && apt-get install -qqy apt-transport-https ca-certificates curl gnupg2 software-properties-common
+RUN curl -fsSL https://download.docker.com/linux/debian/gpg | apt-key add -
+RUN add-apt-repository \
+   "deb [arch=$(dpkg --print-architecture)] https://download.docker.com/linux/debian \
+   $(lsb_release -cs) \
+   stable"
+RUN apt-get update \
```
Comment (Ceiridge) [inline docker/jenkins/Dockerfile:23]: Same as above

**Review by Ceiridge — COMMENTED** [review]
Looks good to me in total, but I have a small suggestion regarding interactivity during apt-get updates

**Review by georgyia — DISMISSED** [review]
TS1, manually tested, worked as expected

File: docker/jenkins/Dockerfile (hunk 1 above)
Comment (reschandreas) [reply docker/jenkins/Dockerfile:16]: good idea

**Review by florian-glombik — APPROVED** [review]
Installation steps look good

**Review by aplr — APPROVED** [review]
Code changes look good, solid DinD solution. Probably document that problems can arise when docker versions deviate too much

**Pull request merged**